src/foraging_gui/Foraging.py

```diff
@@ -204,6 +204,70 @@ def connectSignalsSlots(self):
             for child in container.findChildren((QtWidgets.QLineEdit)):        
                 child.returnPressed.connect(self.keyPressEvent)
     
+    def _warmup(self):
+        '''warm up the session before starting.
+            Use warm up with caution. Usually, it is only used for the first time training. 
+            Turn on the warm up only when all parameters are set correctly, otherwise it would revert 
+            to some incorrect parameters when it was turned off.
+        '''
+        # set warm up parameters
+        if self.warmup.currentText()=='on':
+            # get parameters before the warm up is on
+            self._GetTrainingParameters(prefix='WB_')
```

Comment:
What does `WB_` mean? WarmupBackup? I suggest we use clearer naming conventions to improve readability.

Reply:
Yes, it represents WarmupBackup or WarmupBefore. I make a note to explain the abbreviation and changed the prefix to `WarmupBackup_`